### PR TITLE
Bump WDP revision to address memory leak

### PR DESCRIPTION
**Pull request by remusao**
Resolves https://github.com/brave/brave-browser/issues/23573
Fix in WDP https://github.com/brave/web-discovery-project/pull/191

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by atuchin-m** [comment]
There is a problem with webpack + linkedom.
```
21:06:33  WARNING in ./vendor/web-discovery-project/node_modules/star-wasm/star_wasm_bg.js 250:14-60
21:06:33  Critical dependency: the request of a dependency is an expression
21:06:33   @ ./vendor/web-discovery-project/node_modules/star-wasm/star_wasm.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/hpnv2/star.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/hpnv2/manager.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/hpnv2/background.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app/modules.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app/module.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/index.js
21:06:33   @ ./components/brave_extension/extension/brave_extension/background/webDiscoveryProject.ts
21:06:33   @ ./components/brave_extension/extension/brave_extension/background.ts
21:06:33  
21:06:33  WARNING in ./vendor/web-discovery-project/node_modules/linkedom/commonjs/canvas.cjs
21:06:33  Module not found: Error: Can't resolve 'canvas' in '/home/ubuntu/workspace/pr-brave-browser-bump-wdp-memory-leak-fix-linux/src/brave/vendor/web-discovery-project/node_modules/linkedom/commonjs'
21:06:33   @ ./vendor/web-discovery-project/node_modules/linkedom/commonjs/canvas.cjs
21:06:33   @ ./vendor/web-discovery-project/node_modules/linkedom/esm/html/canvas-element.js
21:06:33   @ ./vendor/web-discovery-project/node_modules/linkedom/esm/shared/html-classes.js
21:06:33   @ ./vendor/web-discovery-project/node_modules/linkedom/esm/index.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/web-discovery-project/html-helpers.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/web-discovery-project/web-discovery-project.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/web-discovery-project/background.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app/modules.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app/module.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/index.js
21:06:33   @ ./components/brave_extension/extension/brave_extension/background/webDiscoveryProject.ts
21:06:33   @ ./components/brave_extension/extension/brave_extension/background.ts
21:06:33  
21:06:33  WARNING in ./vendor/web-discovery-project/node_modules/linkedom/commonjs/perf_hooks.cjs
21:06:33  Module not found: Error: Can't resolve 'perf_hooks' in '/home/ubuntu/workspace/pr-brave-browser-bump-wdp-memory-leak-fix-linux/src/brave/vendor/web-discovery-project/node_modules/linkedom/commonjs'
21:06:33   @ ./vendor/web-discovery-project/node_modules/linkedom/commonjs/perf_hooks.cjs
21:06:33   @ ./vendor/web-discovery-project/node_modules/linkedom/esm/interface/document.js
21:06:33   @ ./vendor/web-discovery-project/node_modules/linkedom/esm/index.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/web-discovery-project/html-helpers.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/web-discovery-project/web-discovery-project.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/web-discovery-project/background.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app/modules.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app/module.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/core/app.js
21:06:33   @ ../out/Static/gen/brave/web-discovery-project/index.js
21:06:33   @ ./components/brave_extension/extension/brave_extension/background/webDiscoveryProject.ts
21:06:33   @ ./components/brave_extension/extension/brave_extension/background.ts
```

**Comment by atuchin-m** [comment]
> There is a problem with webpack + linkedom.
@petemill Could you please take a look, what we can do to support linkedom in WDP?

It seems that updating webpack to v5 could help, but I'm not sure can we do this or not:
https://github.com/cheeriojs/cheerio/issues/2545#issuecomment-1134495836

**Review by diracdeltas — APPROVED** [review]

**Review by atuchin-m — APPROVED** [review]

File: components/webpack/webpack.config.js [hunk 1]
```diff
@@ -119,7 +119,25 @@ module.exports = async function (env, argv) {
         {
           test: /\.(ttf|eot|ico|svg|png|jpg|jpeg|gif)(\?v=[0-9]\.[0-9]\.[0-9])?$/,
           loader: 'file-loader'
-        }]
+        },
+        {
+          test: (input) => {
```
Comment (petemill) [inline components/webpack/webpack.config.js:124]: It doesn't seem clear to me what this rule is for - is it to get around something that works in webpack 5 but not 4? If so, could we leave a comment explaining something like that so we know what to do when we upgrade to webpack 5?

Comment (remusao) [reply components/webpack/webpack.config.js:124]: Good point. I will add a comment.

Comment (remusao) [reply components/webpack/webpack.config.js:143]: @petemill added here: https://github.com/brave/brave-core/pull/13943/commits/acc9d2a1e17932a5261c189735883d92d30413c3

**Review by petemill — APPROVED** [review]

**Review by thypon — APPROVED** [review]

**Pull request merged**